File: Python/Arrays/L_cntNoOfNiceSubstr.py
from typing import List


def numberOfSubarrays(nums: List[int], k: int) -> int:
    n = len(nums)
    cnt = [0] * (n + 1)
    cnt[0] = 1
    ans = 0
    t = 0
    for v in nums:
        t += v & 1
        if t - k >= 0:
            ans += cnt[t - k]
        cnt[t] += 1
    return ans

File: JavaScript/Array/L_cntNoOfNiceSubstr.js
var numberOfSubarrays = function (nums, k) {
    let n = nums.length;
    let cnt = new Array(n + 1).fill(0);
    cnt[0] = 1;
    let ans = 0,
        t = 0;
    for (let v of nums) {
        t += v & 1;
        if (t - k >= 0) {
            ans += cnt[t - k];
        }
        cnt[t]++;
    }
    return ans;
};
